torch/distributed/algorithms/ddp_comm_hooks/optimizer_overlap_hooks.py

```diff
@@ -16,11 +17,10 @@ class _OptimizerHookState(object):
     def __init__(
         self, functional_optim_cls, *functional_optim_args, **functional_optim_kwargs
     ):
-        self.functional_optimizer = functional_optim_cls(
-            [],
+        self.functional_optimizer = create_functional_optim(
```

Comment:
Now that we have a standalone `create_functional_optim` function, I think this `__init__` has become even more redundant. I think it would be cleaner if we simply replaced it with `from_functional_optim`, meaning changing it to `__init__(self, functional_optim)` that contains the current logic of `from_function_optim`.

Reply:
Yep, this is done in the top most diff.